server/src/main/java/io/spine/server/entity/storage/ColumnReader.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2018, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.entity.storage;
+
+import io.spine.annotation.Internal;
+import io.spine.server.entity.Entity;
+
+import java.beans.BeanInfo;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Method;
+import java.util.Collection;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Lists.newLinkedList;
+import static io.spine.server.entity.storage.ColumnRecords.getAnnotatedVersion;
+import static io.spine.util.Exceptions.newIllegalStateException;
+
+/**
+ * A class whose purpose is to obtain {@linkplain EntityColumn entity columns} from the given
+ * {@link Entity} type.
+ *
+ * <p>Each {@code ColumnReader} instance is created for the specified {@link Entity} class.
+ *
+ * <p>Along with obtaining {@link Entity} class columns, the {@code ColumnReader} performs various
+ * checks verifying that {@link EntityColumn} definitions in the processed {@link Entity} class are
+ * correct. If column definitions are incorrect, the exception is thrown upon
+ * {@linkplain EntityColumn entity columns} reading.
+ *
+ * @author Dmytro Kuzmin
+ * @see Columns
+ * @see EntityColumn
+ */
+@Internal
+public class ColumnReader {
+
+    private final Class<? extends Entity> entityClass;
+
+    private ColumnReader(Class<? extends Entity> entityClass) {
+        this.entityClass = entityClass;
+    }
+
+    /**
+     * Creates an instance of {@link ColumnReader} for the given {@link Entity} class.
+     *
+     * <p>The reader can be further used to {@linkplain ColumnReader#readColumns() obtain}
+     * {@linkplain EntityColumn entity columns} for the given class.
+     *
+     * <p>{@code Null} entity class as an argument is not accepted by this method.
```

Comment:
This is a general contract that we have for all the methods by default. See `package-info.java`. 

Please remove this sentence.